sentry_zendesk/client.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import absolute_import, print_function, unicode_literals
+
+from django.utils.encoding import force_bytes  # noqa
+from requests.exceptions import HTTPError
+from sentry.http import build_session
+from sentry_plugins.exceptions import ApiError
+
+from sentry_zendesk import logger
+
+
+class ZendeskClient(object):
+
+    SEARCH_URL = '/api/v2/search.json'
+    CREATE_URL = '/api/v2/tickets.json'
+    HTTP_TIMEOUT = 5
+
+    def __init__(self, zendesk_url, username, password):
+        self.zendesk_url = zendesk_url.rstrip('/')
+        self.username = username
+        self.password = password
+
+    def create_ticket(self, title, comment, ticket_type, problem_id):
+        params = {
+            'ticket': {
+                'type': ticket_type,
+                'subject': title,
+                'comment': comment,
+            }
+        }
+        if problem_id is not None:
+            params['ticket']['problem_id'] = problem_id
+
+        response = self.make_request('post', self.CREATE_URL, params)
+        created_ticket = response.json()['ticket']
+        ticket_id = unicode(created_ticket['id'])
+        logger.info('Created new ticket id "{}"'.format(ticket_id))
+        return ticket_id
+
+    def search_tickets(self, query):
+        params = {'query': 'type:ticket subject:{}*'.format(query)}
```

Comment:
Honestly I don't know either :P
Apparently the results does not respect what is stated in the [documentation](https://developer.zendesk.com/rest_api/docs/core/search).

For example the docs says:

> For example, the search results for "top" will include data that match the string "top" as a single word, a single word in a longer phrase ("top tier"), or the prefix of a longer word ("topology")

But neither ```subject:prob``` nor ```subject:"prob"``` return anything.

Some other tests I made:

- ```subject:problema``` returns results like:

```
problema de licencia 
Problemas em Contatos (A/C Rafael Veturrazzi)
suporte solicitado por email - problemas na analise harmônica
Problemas para executar o Fluent 15.0.0 no cluster
Problemas em resultados de simulação com contato
```

- ```subject:problema licencia``` returns results like:

```
Problema de instalacion de licencia
Problema con instalación de licencia
Problemas de reconocimiento de mallas en  fluent
Problemas de licencia
```
- ```subject:problema licen``` returns nothing.
- ```subject:problema licen*``` returns nothing.
- ```subject:problema licenc``` returns some results but much less than ```subject:problema licencia```.
- ```subject:prob``` returns nothing
- ```subject:prob*``` returns results with the word problema anywhere

So, when there is only one word the wildcard makes difference. However with multiple word the wildcard does not help and only querying the whole words seems reliable


```
```